Unicode/ustring.c

```diff
@@ -562,12 +562,12 @@ void utf82u_strcat(unichar_t *to,const char *from) {
     utf82u_strcpy(to+u_strlen(to),from);
 }
 
-char *u2utf8_strcpy(char *utf8buf,const unichar_t *ubuf) {
+char *u2utf8_strncpy(char *utf8buf,const unichar_t *ubuf,int len) {
 /* Copy unichar string 'ubuf' into utf8 buffer string 'utf8buf' */
     char *pt = utf8buf;
 
     if ( ubuf!=NULL ) {
-	while ( *ubuf && (pt=utf8_idpb(pt,*ubuf++,0)) );
+	while ( *ubuf && (pt=utf8_idpb(pt,*ubuf++,0)) && --len );
 	if ( pt ) {
```

Comment:
I dunno what the thought behind this was, but `utf8_idpb()` assigns to the argument corresponding to `pt`, and then increments, without checking whether its null. So this conditional seems useless to me -- it's not like the pointer is likely to wrap around to zero in just the right place. 

Better to assert utf8buf isn't at the top (for debug) and remove the conditional.

Reply:
no -- assert *`utf8buf`* is not NULL, not `ubuf`. Leave the outer guard on `ubuf`, it might indeed be null. It's the inner `pt` guard that I think is useless.

Reply:
Sorry, I'll fix it :)

Reply:
This time I looked at it longer, I think it's what you're asking for now